### PR TITLE
[IOTDB-1156] Improve the quality and efficiency of code formatting

### PR DESCRIPTION
**Pull request by sunjincheng121**
## Description
This PR includes two commits, one is the POM configuration of the Spotless plug-in, the other is format the code by using spotless / Google Java format. 

Note: 
- We also need a PR (Web site) to explain how the IDE is configured, which will be opened in a separate PR #2686 .
- When this PR is merged, other PRS can be merged according to the following steps(take pull/2667 for example):
`asf` is repo of Apache IoTDB
 ```
gf asf pull/2667/head:PR2667
gck PR2667
git cherry-pick 6aedcdbefe37f929300d032eec29df4d3465dcac 
mvn spotless:check
mvn spotless:apply
git add .
git commit -m 'Apply spotless'
git rebase asf_master
// Resolve your conflict
git rebase --continue
// After squash your commits
gck asf_master
git merge PR2667
git push asf asf_master:master

```
In the subsequent PR, only needs to run ` mvn spotless:check` And `mvn spotless:apply` to ensure that our code format will always be correct and unified, and we can integrate Spotless in CI, so that our contributors will never care about code format.

### CODE REVIEW

**Comment by sunjincheng121** [comment]
Add one commit about upgrade the sonar-maven-plugin version to 3.7.0.1746 due to ubuntu(11) CI check error: https://github.com/apache/iotdb/runs/1904211046#step:8:11596

**Comment by sunjincheng121** [comment]
I have checked this CI error, and ensure that this error is not caused by the current PR. the same problem occurred in [1][2]. I created JIRA[3] to focus on tracking.
[1] https://github.com/apache/iotdb/pull/2677/checks?check_run_id=1892946112
[2] https://github.com/apache/iotdb/pull/2667/checks?check_run_id=1863611951
[3] https://issues.apache.org/jira/browse/IOTDB-1157

**Comment by jixuan1989** [comment]
> https://issues.apache.org/jira/browse/IOTDB-1157

It is not caused by sonar version.

If you want to avoid warnings, see [1] , by using 
```
- uses: actions/checkout@v2
  with:
    fetch-depth: 0
```

If you want to solve NullPointer issues, I have no idea. If you check the sonar command, you will find the problem is `sonar.login` is null, 
image

see [2] and [3] for why. and that is why I open PR #2681 

[1] https://community.sonarsource.com/t/sonarcloud-with-github-actions/22713
[2] https://www.google.com/url?q=https://help.github.com/en/articles/events-that-trigger-workflows%23pull-request-events-for-forked-repositories&source=gmail&ust=1613530497486000&usg=AFQjCNEfebNVY_hNI4Gtz41Cba39YTgLhw
[3] https://lists.apache.org/thread.html/rffca7a170b1e982c050279bf80802660b2c18131963b34c22122fb95%40%3Cbuilds.apache.org%3E

**Comment by sunjincheng121** [comment]
> > https://issues.apache.org/jira/browse/IOTDB-1157
> 
> It is not caused by sonar version.
> 
> If you want to avoid warnings, see [1] , by using
> 
> ```
> - uses: actions/checkout@v2
>   with:
>     fetch-depth: 0
> ```
> 
> If you want to solve NullPointer issues, I have no idea. If you check the sonar command, you will find the problem is `sonar.login` is null,
> image
> 
> see [2] and [3] for why. and that is why I open PR #2681
> 
> [1] https://community.sonarsource.com/t/sonarcloud-with-github-actions/22713
> [2] https://www.google.com/url?q=https://help.github.com/en/articles/events-that-trigger-workflows%23pull-request-events-for-forked-repositories&source=gmail&ust=1613530497486000&usg=AFQjCNEfebNVY_hNI4Gtz41Cba39YTgLhw
> [3] https://lists.apache.org/thread.html/rffca7a170b1e982c050279bf80802660b2c18131963b34c22122fb95%40%3Cbuilds.apache.org%3E

Opened the PR for Correct the config of `sonar.java.checkstyle.reportPaths`. https://github.com/apache/iotdb/pull/2685

**Comment by sunjincheng121** [comment]
should add one improvement fixup commit later before merged.

**Comment by sunjincheng121** [comment]
I have update the PR by the follows changes:
- Fix incorrect license headers, i.e., `/**` to `/*` in License header.
- Turn special class comments into multiline comments for ensure example readability.

**Comment by sunjincheng121** [comment]
Rebase code from master...

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java [hunk 1]
```diff
@@ -18,18 +18,21 @@
...
+import java.util.List;
+import java.util.Map;
+
 public class AlignByDevicePlan extends QueryPlan {
 
-  private List<String> measurements; // to record result measurement columns, e.g. temperature, status, speed
-  private Map<String, String> measurementAliasMap; // select s1, s2 as speed from root, then s2 -> speed
+  private List<String>
+      measurements; // to record result measurement columns, e.g. temperature, status, speed
+  private Map<String, String>
```
Comment (liutaohua) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java:34]: Don't use end-of-line comments?

**Review by liutaohua — APPROVED** [review]
LGTM

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java (hunk 1 above)
Comment (sunjincheng121) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java:34]: At present, we have a lot of end of line comments, which are very convenient. If we want to prohibit the use of end of line comments, we need to initiate a discussion. due to we 
 may  need  modify Google-Java-format. I suggest solving it in another PR after discussion. What do you think?

File: pom.xml [hunk 2]
```diff
@@ -598,6 +599,34 @@
...
+                    <version>${spotless.version}</version>
+                    <configuration>
+                        <java>
+                            <googleJavaFormat>
+                                <version>1.7</version>
+                                <style>GOOGLE</style>
+                            </googleJavaFormat>
+
+                            <importOrder>
+                                <order>org.apache.iotdb,,javax,java,\#</order>
```
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply pom.xml:614]: This affects the order in which packages are imported. 

```
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.apache.commons.io.FileUtils;
```
vs 

```
import org.apache.commons.io.FileUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
```
We are in the latter order at present. :)

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java (hunk 1 above)
Comment (liutaohua) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java:34]: okay

**Comment by sunjincheng121** [comment]
Thanks for your review @liutaohua !
I have update the PR accordingly in `fixup` commit. and rebase the code in another commit.

**Comment by sunjincheng121** [comment]
I would like to do the last check before merging...

**Comment by sunjincheng121** [comment]
Thanks for your review @liutaohua 
I have update the PR accordingly in `fixup` commit. and rebase the code in another commit.

**Comment by sunjincheng121** [comment]
I have squash the commits, and Keep 2 commits due to the first commit can help to rebase other PRs.

BTW: I'd like to add .git-blame-ignore-revs for ignore this big code format refactor after this PR has been merged.

**Comment by sunjincheng121** [comment]
I have squash the commits, and Keep 2 commits due to the first commit can help us to rebase other PRs.
BTW: I'd like to add .git-blame-ignore-revs for ignore this big code format refactor after this PR has been merged.

**Pull request merged**